bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_gnn.py

```diff
@@ -0,0 +1,259 @@
+                    make_mlp_model(gin, edge_model._layers[0], gbl_namespace.TMVA.Experimental.SOFIE.FunctionTarget.EDGES, gbl_namespace.TMVA.Experimental.SOFIE.GraphType.GNN)
+                elif(i.name == 'layer_norm'):
+                    add_layer_norm(i, gin.edges_update_block)
+                else:
+                    print("Invalid Model for edge update.")
+                    return
+        
+        else:
+            print("Invalid Model for edge update.")
+            return
+
+        add_weights(edge_model.variables, gin.edges_update_block)
+
+        # adding the global update function
+        global_model = GraphModule._global_block._global_model
+        if (global_model.name == 'mlp'):
+            make_mlp_model(gin, global_model, gbl_namespace.TMVA.Experimental.SOFIE.FunctionTarget.GLOBALS, gbl_namespace.TMVA.Experimental.SOFIE.GraphType.GNN)
+        elif (global_model.name == 'sequential'):
+            for i in global_model._layers:
+                if(i.name == 'mlp'):
+                    make_mlp_model(gin, global_model._layers[0], gbl_namespace.TMVA.Experimental.SOFIE.FunctionTarget.GLOBALS, gbl_namespace.TMVA.Experimental.SOFIE.GraphType.GNN)
+                elif(i.name == 'layer_norm'):
+                    add_layer_norm(i, gin.globals_update_block)
+                else:
+                    print("Invalid Model for global update.")
+                    return
+        
+        else:
+            print("Invalid Model for global update.")
+            return
+
+        add_weights(global_model.variables, gin.globals_update_block)
+
+        # adding edge-node aggregate function
+        add_aggregate_function(gin, GraphModule._node_block._received_edges_aggregator._reducer.__qualname__, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.NODES_EDGES)
+
+        # adding node-global aggregate function
+        add_aggregate_function(gin, GraphModule._global_block._nodes_aggregator._reducer.__qualname__, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.NODES_GLOBALS)
+
+        # adding edge-global aggregate function
+        add_aggregate_function(gin, GraphModule._global_block._edges_aggregator._reducer.__qualname__, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.EDGES_GLOBALS)
+
+        gnn_model = gbl_namespace.TMVA.Experimental.SOFIE.RModel_GNN(gin)
+        blas_routines = gbl_namespace.std.vector['std::string']()
+        blas_routines.push_back("Gemm")
+        blas_routines.push_back("Axpy")
+        blas_routines.push_back("Gemv")
+        gnn_model.AddBlasRoutines(blas_routines)
+        return gnn_model
+
+class RModel_GraphIndependent:
+    def ParseFromMemory(GraphModule, GraphData, filename = "graph_independent_network"):
+        gin = gbl_namespace.TMVA.Experimental.SOFIE.GraphIndependent_Init()
+        gin.num_nodes = len(GraphData['nodes'])
+
+        # extracting the edges
+        edges = []
+        for i in range(len(GraphData['senders'])):
```

Comment:
the `for i in range(len(iterable))` idiom is not really Pythonic. It should be `for item in iterable`. Please change it.

Reply:
made the change, thanks!